test/common/testlib.py

```diff
@@ -843,40 +842,41 @@ def check_issue(self, trace):
             traceback.print_exc()
         return number
 
-    def check_retry(self, trace):
-        #
+    def check_retry(self, trace, tries):
+
+        # Never try more than five times
+        if tries >= 5:
+            return False
+
         # We check for persistent but test harness or framework specific
         # failures that otherwise cause flakiness and false positives.
         #
         # The things we check here must:
         #  * have no impact on users of Cockpit in the real world
         #  * be things we tried to resolve in other ways. This is a last resort
         #
-        retry = False
-        if self.retryable:
-            trace = self.normalize_traceback(trace)
-
-            # HACK: An issue in phantomjs and QtWebkit
-            # http://stackoverflow.com/questions/35337304/qnetworkreply-network-access-is-disabled-in-qwebview
-            # https://github.com/owncloud/client/issues/3600
-            # https://github.com/ariya/phantomjs/issues/14789
-            if "PhantomJS or driver broken" in trace:
-                retry = True
-
-            # HACK: Interacting with sshd during boot is not always predictable
-            # We're using an implementation detail of the server as our "way in" for testing.
-            # This often has to do with sshd being restarted for some reason
-            if "SSH master process exited with code: 255" in trace:
-                retry = True
-
-        if retry:
-            self.retryable = False
-        return retry
+        trace = self.normalize_traceback(trace)
+
+        # HACK: An issue in phantomjs and QtWebkit
+        # http://stackoverflow.com/questions/35337304/qnetworkreply-network-access-is-disabled-in-qwebview
+        # https://github.com/owncloud/client/issues/3600
+        # https://github.com/ariya/phantomjs/issues/14789
+        if "PhantomJS or driver broken" in trace:
+            return True
+
+        # HACK: Interacting with sshd during boot is not always predictable
+        # We're using an implementation detail of the server as our "way in" for testing.
+        # This often has to do with sshd being restarted for some reason
+        if "SSH master process exited with code: 255" in trace:
+            return True
+
+        return False
 
 class TapResult(unittest.TestResult):
     def __init__(self, stream, descriptions, verbosity):
         self.offset = 0
         self.policy = None
+        self.tries = 1
```

Comment:
The same TapResult is used for different tests in a TestSuite. But I guess we can limit to 5 retries per test suite for now. I'd be fine with that.